### ISSUE TITLE
Zod-OpenAPI no longer works with Hono 4.3.0

### ISSUE THREAD

**Issue by bruceharrison1984** [honojs/middleware#496]
After upgrading Hono to 4.3.0, OpenAPI no longer appears to match response types.

![image](https://github.com/honojs/middleware/assets/17836372/09c91c6a-dbf7-4d68-b6e6-0e98971f3852)

```ts
export const userRoutes = app.openapi(
  createRoute({
    method: 'get',
    path: '/',
    tags: ['user'],
    security: [{ Bearer: [] }],
    request: {
      query: z.object({ limit: z.number(), offset: z.number() }),
    },
    responses: {
      200: {
        content: {
          'application/json': {
            schema: z.array(selectSchema),
          },
        },
        description: 'list of user',
      },
    },
  }),
  async (c) => {
    const { limit, offset } = c.req.queries();
    const resource = await c.var.MY_REPOSITORY.user.getAll(
      Number.parseInt(limit[0]),
      Number.parseInt(offset[0])
    );
    return c.json(resource, 200);
  }
);
```

Downgrading Hono back to `4.2.5` fixes the types again:
![image](https://github.com/honojs/middleware/assets/17836372/72da315b-6287-4ca2-8806-1445e1979979)

**Comment by yusukebe** [comment]
@bruceharrison1984 

Thanks! We have to fix this issue.

Hi @NamesMT, are you interested in this Zod OpenAPI? If so, please check this issue. But I will be able to take a look later.

**Comment by NamesMT** [comment]
Seems to be caused since `8129a3a - (4 days ago) feat(rpc): Add status code to response type`

**Comment by NamesMT** [comment]
Okay, so on `4.3.0` we have two types breaking on `ToSchema`:
  - Update from `O` is output to `extends TypedResponse` - 8129a3a
  - Update from `I` accepting `Input['in']` to `Input` - ea4e998

Previously I thought `ToSchema` is only for internal use so I didn't think much about it, but if it is exposed for other libraries I think we should revert `ToSchema` and creates a new `ToSchemaV2` for future uses to revert the breaking changes?

**Comment by NamesMT** [comment]
I will try to make a fast hotfix first to introduce some compatibility with older version that we can quick release.
Then we can create an issue to discuss the matter above ^

**Comment by yusukebe** [comment]
@NamesMT 

Okay, thanks!

**Comment by yusukebe** [comment]
Hi @bruceharrison1984 

Thanks to @NamesMT , this has been fixed with the new version `4.3.1`. Try it.
